prepare-commit-msg

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+# WordPress prepare-commit-msg hook
+
+DEV_LIB_COMMIT_MESSAGE_FILE="/tmp/.dev-lib-commit-message.$PPID"
+COMMIT_EDITMSG=$1
+
+JIRA_TASK_REGEXP='.*([A-Z]{1,32}-[0-9]{1,32}).*'
+JIRA_BRANCH_REGEXP='([a-zA-Z]+-[0-9]{1,32})';
+if [[ ! $(cat "$COMMIT_EDITMSG") =~ $JIRA_TASK_REGEXP ]]; then
+	JIRA_TASK=$(git rev-parse --abbrev-ref HEAD | grep -oP "$JIRA_BRANCH_REGEXP")
```

Comment:
Also, make sure to include the `-E` param to indicate that extended regular expressions are to be used in `grep`. Otherwise, the `+` repeater and bracket notation don't work on OSX.